ecal/core/src/ecal_registration_provider.cpp

```diff
@@ -320,7 +321,18 @@ namespace eCAL
     std::lock_guard<std::mutex> lock(m_server_map_sync);
     for(SampleMapT::const_iterator iter = m_server_map.begin(); iter != m_server_map.end(); ++iter)
     {
-      // register sample
+      //////////////////////////////////////////////
+      // update description
+      //////////////////////////////////////////////
+      auto& ecal_sample_service = iter->second.service();
+      for (auto method : ecal_sample_service.methods())
```

Comment:
I guess this can be made a const reference? Please double check.

```suggestion
      for (const auto& method : ecal_sample_service.methods())
```